packages/flutter/lib/src/material/page.dart

```diff
@@ -32,7 +32,7 @@ import 'theme.dart';
 ///  * [MaterialRouteTransitionMixin], which provides the material transition
 ///    for this route.
 ///  * [MaterialPage], which is a [Page] of this class.
-class MaterialPageRoute<T> extends PageRoute<T> with MaterialRouteTransitionMixin<T> {
+class MaterialPageRoute<T> extends PageRoute<T> with MaterialRouteTransitionMixin<T>, FlexibleTransitionRouteMixin<T> {
```

Comment:
The use of FlexibleTransitionRouteMixin won't be necessary when that code is moved into ModalRoute, and any other people that extended ModalRoute will get the new functionality automatically 👍 .